src/mcl_3dl.cpp

```diff
@@ -1191,6 +1191,87 @@ class MCL3dlNode
     return true;
   }
 
+  void configureFilter()
```

Comment:
Is this intended to be called multiple times on dynamic reconfigure?
Some of them (e.g. resetting the `FilterVec3`s) are unsafe to do while running.

Reply:
I reverted the changes where I split `configure` and `configureFilter`. Let's discuss more in depth the design for dynamic parameters separately. This PR mainly focuses on providing better isolation for ROS parameter server access.